python/dolfinx/common.py

```diff
@@ -5,35 +5,47 @@
 # SPDX-License-Identifier:    LGPL-3.0-or-later
 """General tools for timing and configuration."""
 
+from mpi4py import MPI as _MPI
+
 import functools
 import typing
 
 from dolfinx import cpp as _cpp
-from dolfinx.cpp.common import (IndexMap, git_commit_hash, has_adios2,  # noqa
-                                has_debug, has_kahip, has_parmetis)
+from dolfinx.cpp.common import has_adios2  # noqa
+from dolfinx.cpp.common import (IndexMap, git_commit_hash, has_debug,  # noqa
+                                has_kahip, has_parmetis)
 
-__all__ = ["IndexMap", "Timer", "timed"]
+__all__ = ["IndexMap", "Timer", "timed", "list_timings", "timing", "TimingType", "Reduction"]
 
 TimingType = _cpp.common.TimingType
 Reduction = _cpp.common.Reduction
 
 
 def timing(task: str):
+    """Return timing (count, total wall time, total user time, total system time) for given task."""
     return _cpp.common.timing(task)
 
 
-def list_timings(comm, timing_types: list, reduction=Reduction.max):
-    """Print out a summary of all Timer measurements, with a choice of
-    wall time, system time or user time. When used in parallel, a
-    reduction is applied across all processes. By default, the maximum
-    time is shown."""
+def list_timings(comm: _MPI.Comm, timing_types, reduction=Reduction.max):
+    """Print out a summary of all Timer measurements.
+
+    One can specify the timing types as a subset of wall time, system time or user time.
+
+    Note:
+        When used in parallel, a reduction is applied across all processes. By default, the maximum time is shown.
+
+    Args:
+        comm: Communicator to reduce timings over
+        timing_types: List of timings to return
```

Comment:
What's the type (e.g., list of enums?) and what are the possible values?

Reply:
Could not use type hints due to nanobind missing stubs (so mypy failed).
Can add a reference to the cpp documentation